app/src/main/java/com/android/partagix/ui/screens/HomeScreen.kt

```diff
@@ -1,40 +1,149 @@
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.android.partagix.model.HomeViewModel
+import com.android.partagix.model.InventoryViewModel
 import com.android.partagix.ui.components.BottomNavigationBar
+import com.android.partagix.ui.components.ItemListColumn
 import com.android.partagix.ui.navigation.NavigationActions
 import com.android.partagix.ui.navigation.Route
 
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun HomeScreen(navigationActions: NavigationActions, modifier: Modifier = Modifier) {
+fun HomeScreen(
+    homeViewModel: HomeViewModel,
+    inventoryViewModel: InventoryViewModel,
+    navigationActions: NavigationActions,
+    modifier: Modifier = Modifier
+) {
+  val uiState by inventoryViewModel.uiState.collectAsStateWithLifecycle()
+  inventoryViewModel.getInventory()
   Scaffold(
       modifier = modifier.testTag("homeScreen"),
+      topBar = {
+        TopAppBar(
+            modifier = Modifier.testTag("homeScreenTopAppBar"),
+            title = { Text(text = "Welcome back, ${"User" /* TODO get actual user name */}") },
+            actions = {
+              IconButton(onClick = { /* TODO go to notification screen */}) {
+                Icon(Icons.Default.Notifications, contentDescription = "Notifications")
+              }
+            })
+      },
       bottomBar = {
         BottomNavigationBar(
             selectedDestination = Route.HOME,
             navigateToTopLevelDestination = navigationActions::navigateTo,
-            modifier = modifier.testTag("homeScreenBottomNavBar"))
+            modifier = Modifier.testTag("homeScreenBottomNavBar"))
       }) { innerPadding ->
         Column(
-            modifier = Modifier.fillMaxSize().padding(innerPadding),
-            verticalArrangement = Arrangement.Center,
-            horizontalAlignment = Alignment.CenterHorizontally) {
+            modifier =
+                Modifier.fillMaxSize().padding(innerPadding).testTag("homeScreenMainContent")) {
+              HorizontalDivider(modifier = Modifier.fillMaxWidth())
               Text(
-                  text = "Home Screen coming soon!",
-                  style = TextStyle(fontFamily = FontFamily.Monospace, fontSize = 24.sp),
-                  modifier = Modifier.testTag("homeScreenMainContent"))
+                  text = "Quick access",
+                  modifier = Modifier.fillMaxWidth().padding(start = 16.dp, top = 16.dp),
+                  style = MaterialTheme.typography.titleLarge)
+              Spacer(modifier = Modifier.height(8.dp))
+              Row(
+                  modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 8.dp),
+                  horizontalArrangement = Arrangement.SpaceBetween) {
+                    BigButton(
+                        logo = Icons.Default.PersonSearch,
+                        text = "Find item to borrow",
```

Comment:
You could create const or var fields to put the text strings. This could make further changes easier, as well as allow translation in the future.

Reply:
also addressed in the following commits